enzy_htp/analysis/rmsd.py

```diff
@@ -0,0 +1,120 @@
+    
+    Args:
+        mask_region (List[Residue], optional): A list of Residues to calculation RMSD value. If the list is empty, then all residues are selected.
+        interface (str, optional): The interface for calculation. Default amber.
+        include_ligand (bool, optional): Indicate if ligands are included in RMSD calculation. Default True.
+        ca_only (bool, optional): Indicate if only C-alpha are included in RMSD calculation; otherwise all atoms except hydrogens are included. Default True.
+    """
+    interface = interface.lower()
+    mask_pattern = str()
+    
+    if (interface == "amber"):
+        if (mask_region):
+            mask_pattern = f":{','.join([str(residue.idx) for residue in mask_region])}"
+        else:
+            mask_pattern = ""
+        if (ca_only):
+            if mask_pattern:
+                mask_pattern += "&"
+            mask_pattern += "@CA"
+            
+    elif (interface == "pymol"):
+        if (mask_region):
+            mask_pattern = f"resi {'+'.join([str(residue.idx) for residue in mask_region])}"
+        else:
+            mask_pattern = "all"
+        if (ca_only):
+            mask_pattern += " and name CA"
+
+    elif (interface not in ACCEPTED_INTERFACES.keys()):
+        raise ValueError(f"Unacceptable interface value. Please choose from {' and '.join(ACCEPTED_INTERFACES.keys())}")
+    return mask_pattern
+
+def rmsd_of_structure(structure_ensemble: StructureEnsemble, interface: str = "amber", reference_structure: Structure = None, include_ligand: bool = True, ca_only: bool = True) -> float:
+    """Get RMSD value from MD simulation result of the whole structure.
+    
+    Args:
+        structure_ensemble (StructureEnsemble): A collection of different geometries of the same enzyme structure.
+        interface (str, optional): The interface for calculation. Default amber.
+        reference_structure (Structure, optional): The reference structure, which should have the same sequences and ligangs as the structure_ensemble. Default: Average structure.
+        include_ligand (bool, optional): Indicate if ligands are included in RMSD calculation. Default True.
+        ca_only (bool, optional): Indicate if only C-alpha are included in RMSD calculation; otherwise all atoms except hydrogens are included. Default True.
+    """
+    interface = interface.lower()
+    mask_region = structure_ensemble.structure_0.residues
+    if (not include_ligand):
+        mask_region = list(filter(lambda residue: (not residue.is_noncanonical()), mask_region))
+    return rmsd_of_region(
+        structure_ensemble=structure_ensemble, 
+        interface=interface,
+        reference_structure=reference_structure, 
+        mask_region=mask_region, 
+        ca_only=ca_only
+    )
+
+def rmsd_of_region(structure_ensemble: StructureEnsemble, interface: str = "amber", reference_structure: Structure = None, mask_region: List[Residue] = list(), ca_only: bool = True) -> float:
```

Comment:
This seems the core API in this module. However, it is limited to receive input as a list of Residue()s. However, we need to support any subregion of a protein even if they are not full residues.